osquery/tables/system/windows/user_groups.cpp

```diff
@@ -0,0 +1,94 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <Windows.h>
+// clang-format off
+#include <LM.h>
+// clang-format on
+
+#include <osquery/core.h>
+#include <osquery/tables.h>
+#include <osquery/logger.h>
+#include <osquery/sql.h>
+
+#include "osquery/core/process.h"
+#include "osquery/core/windows/process_ops.h"
+#include "osquery/core/windows/wmi.h"
+#include "osquery/tables/system/windows/registry.h"
+#include "osquery/core/conversions.h"
+
+namespace osquery {
+
+std::string psidToString(PSID sid);
+int getGidFromSid(PSID sid);
+
+namespace tables {
+
+void processLocalUserGroups(std::string uid,
+                            std::string user,
+                            QueryData& results) {
+  unsigned long dwUserGroupInfoLevel = 0;
+  unsigned long numGroups = 0;
+  unsigned long dwTotalUserGroupMembers = 0;
+  LOCALGROUP_USERS_INFO_0* ginfo = nullptr;
+  PSID sid = nullptr;
+
+  unsigned long ret = 0;
+  do {
+    ret = NetUserGetLocalGroups(nullptr,
+                                stringToWstring(user).c_str(),
+                                dwUserGroupInfoLevel,
+                                1,
+                                (LPBYTE*)&ginfo,
+                                MAX_PREFERRED_LENGTH,
+                                &numGroups,
+                                &dwTotalUserGroupMembers);
+
+    if ((ret != NERR_Success && ret != ERROR_MORE_DATA) || ginfo == nullptr) {
+      VLOG(1) << " NetUserGetLocalGroups failed for user " << user << " with "
+              << ret;
+      continue;

```

Comment:
Shouldn't this be a break?

Reply:
Wow, this whole `do/while` loop should be removed actually. I had a different thought about what `NetUserGetLocalGroups` was doing. Thanks!